var/spack/repos/builtin/packages/atlas/package.py

```diff
@@ -107,28 +107,23 @@ def install(self, spec, prefix):
             make("install")
             self.install_test()
 
-    def setup_dependent_package(self, module, dspec):
-        # libsatlas.[so,dylib,dll ] contains all serial APIs (serial lapack,
-        # serial BLAS), and all ATLAS symbols needed to support them. Whereas
-        # libtatlas.[so,dylib,dll ] is parallel (multithreaded) version.
-        name = 'libsatlas.%s' % dso_suffix
-        libdir = find_library_path(name,
-                                   self.prefix.lib64,
-                                   self.prefix.lib)
-
-        if '+shared' in self.spec:
-            self.spec.blas_shared_lib   = join_path(libdir, name)
-            self.spec.lapack_shared_lib = self.spec.blas_shared_lib
+    @property
+    def blas_libs(self):
+        shared = True if '+shared' in self.spec else False
+        return find_libraries(['libsatlas'], root=self.prefix, shared=shared, recurse=True)
```

Comment:
surprisingly it took 6 mins for me with gcc on Ubuntu 16 LTS. As for libs: shared are fine as in my comment above. For static the single threaded list is above, multithreaded is
```
['liblapack' 'libptcblas' 'libptf77blas' 'libatlas']
```
Here's summary on static libs: 
```
liblapack.a	LAPACK
libcblas.a	The ANSI C interface to the BLAS
libf77blas.a	The Fortran77 interface to the BLAS
libptcblas.a	The ANSI C interface to the threaded (SMP) BLAS
libptf77blas.a	The Fortran77 interface to the threaded (SMP) BLAS
libatlas.a	The main ATLAS library, providing low-level routines for all interface libs
```